### PR TITLE
Refactor AuthKeyPair.raw_create_key_pair

### PR DESCRIPTION
**Pull request by agrare**
Merge together with: https://github.com/ManageIQ/manageiq/pull/21759

### CODE REVIEW

File: app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb [hunk 1]
```diff
@@ -1,6 +1,4 @@
 class ManageIQ::Providers::Amazon::CloudManager::AuthKeyPair < ManageIQ::Providers::CloudManager::AuthKeyPair
-  AwsKeyPair = Struct.new(:name, :key_name, :fingerprint, :private_key)
```
Comment (Fryguy) [inline app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb:2]: Was key_name just not used?

Comment (agrare) [reply app/models/manageiq/providers/amazon/cloud_manager/auth_key_pair.rb:2]: So I'm guessing they created `AwsKeyPair` to match the Openstack object as closely as possible, but key_name was not used by https://github.com/ManageIQ/manageiq/pull/21759/files#diff-b80f64b28488206bf8d41b9b53fb77a5490267a1a91b5d1e4f0897af88951554L43-L46 so I didn't include it.  This is now just returning the params for creating of the record

**Review by Fryguy — APPROVED** [review]

**Comment by kbrock** [comment]
- I like the moving away from using an openstruct.
- Would it make sense for `raw_create_*` actually create the object instead of creating parameters? Maybe this is a trend we use elsewhere. sorry for the lazy question

**Comment by agrare** [comment]
> Would it make sense for raw_create_* actually create the object instead of creating parameters? Maybe this is a trend we use elsewhere. sorry for the lazy question

The create_key_pair operation is unique in creating a database record, because if a private key is generated it isn't possible to fetch it again as part of refresh so we have to create the db record here.  The raw_* methods typically do the provider specific bits and creating the db record isn't provider specific which is why I left it out of here.

**Comment by Fryguy** [comment]
Backported to `najdorf` in commit c94d01f4acf4581b76cc755afe9589cbd4f5b63d.

```text
commit c94d01f4acf4581b76cc755afe9589cbd4f5b63d
Author: Jason Frey <fryguy9@gmail.com>
Date:   Tue Mar 8 14:45:48 2022 -0500

    Merge pull request #746 from agrare/refactor_create_key_pair
    
    Refactor AuthKeyPair.raw_create_key_pair
    
    (cherry picked from commit eab5fa966b6fc7553b1ad0a110e444996dc2b795)
```

**Pull request merged**